File: Baekjoon/이분탐색/H_ThreeDots_이현정.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws Exception {
		System.setIn(new FileInputStream("src/input.txt"));
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int T = Integer.parseInt(br.readLine());
		
		for(int i = 0; i < T; i++) {
			int N = Integer.parseInt(br.readLine());
			StringTokenizer stk = new StringTokenizer(br.readLine());
			
			int[] arr = new int[N];
			
			for(int j = 0; j < N; j++) {
				arr[j] = Integer.parseInt(stk.nextToken());
			}
			
			int res = 0;
			
			HashMap<Integer, Boolean> codes = new HashMap<Integer, Boolean>(N);
			
			for(int z = 0; z < N; z++) {
				codes.put(arr[z], true);
			}
			
			Arrays.sort(arr);
			
			for(int a = 0; a < N; a++) {
				for(int b = a+1; b < N; b++) {
					int c = arr[b]*2 - arr[a];
					if (codes.get(c) != null) {
						res += 1;
				}
			}
			}
			
			System.out.println(res);
			
		}
	}
}

File: Baekjoon/이분탐색/H_ThreeDots_이현정.py
import sys
sys.stdin = open('input.txt')

T = int(sys.stdin.readline())
for _ in range(T):
    N = int(sys.stdin.readline())
    dot = sorted(list(map(int, sys.stdin.readline().split())))
    res = 0

    d = {i: True for i in dot}

    for i in range(N):
        for j in range(i+1, N):
            if (dot[j]*2 - dot[i]) in d:
                res += 1
    print(res)
